src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1542,20 +1542,22 @@ export class SavedObjectsRepository {
   }
 
   /**
-   * Increases a counter field by one. Creates the document if one doesn't exist for the given id.
+   * Increases a counter field by incrementValue which by default is 1. Creates the document if one doesn't exist for the given id.
    *
    * @param {string} type
    * @param {string} id
    * @param {string} counterFieldName
    * @param {object} [options={}]
+   * @param {number} [incrementValue=1]
    * @property {object} [options.migrationVersion=undefined]
    * @returns {promise}
    */
   async incrementCounter(
     type: string,
     id: string,
     counterFieldName: string,
-    options: SavedObjectsIncrementCounterOptions = {}
+    options: SavedObjectsIncrementCounterOptions = {},
```

Comment:
these options look interesting.

seems like it has

```ts
    // (undocumented)
    migrationVersion?: SavedObjectsMigrationVersion;
    refresh?: MutatingOperationRefreshSetting;
}
```

qq: do we know what this refresh is. is it worth stuff this increment interval within `options`?

Reply:
I investigated the `options`. 
The `refresh` options used here is passed down to savedObject client and refers to an option which tells OpenSearch to make the operation visible to searching. Please find more details about it [here](https://opensearch.org/docs/latest/api-reference/document-apis/update-document/#upsert).

The `migrationVersion` settings is used to migrate document to its latest version.

Overall both these options dictate how the object gets saved in openSearch. 

Adding incrementInterval as an `option` may make things more complicated. Currently the increment Counter is only being used to govern UI stats other telemetry information of App Usage and User Agent are not using this method. Therefore we will need to add a batching mechanism separate for the App Usage and User Agent. This makes the idea it more complicated than the current approach where we do batching on report level without being worried about its internal contents

Reply:
ty for the insight